docs/architecture/app-class-and-template.md

```diff
@@ -0,0 +1,200 @@
+# ApplicationClass and ApplicationTemplate
+
+Managing Applications can be passed on some external system, to remove the burden of integration with Compass from Applications.
+Because of that, we are going to extend our API with IntegrationSystem and ApplicationClass types.
+
+To increase the reusability of the application configuration, we introduce ApplicationTemplate. 
+
+
+## Terminology
+
+- **Application** represents functionality that provides API or Events.
+
+- **ApplicationClass** represents soon-to-be Application that is manged by IntegrationSystem. 
+   
+- **ApplicationTemplate** represents soon-to-be Application with all details already provided (like API and Events) that 
+can be created by providing values for replacing placeholders.
+  
+- **IntegrationSystem** (abbreviation: **IntSys**):
+
+    Any system that works in the context of multiple tenants, managing multiple Applications or Runtimes. It exists in a global scope without a tenant assignment.
+
+## GraphQL API Changes
+
+1. Managing Integration Systems
+ 
+```
+type IntegrationSystem {
+    id: ID
+    name: String
+    description: String
+    auths: [SystemAuth]
+}
+
+input IntegrationSystemInput {
+    name: String!
+    description: String
+}
+
+type Mutation {
+    createIntegrationSystem(in: IntegrationSystemInput!): IntegrationSystem!
+    updateIntegrationSystem(id: ID!, in: IntegrationSystemInput!): IntegrationSystem!
+    deleteIntegrationSystem(id: ID!): IntegrationSystem! 
+    
+    addCredentialsForIntegrationSystem(id: ID!): SystemAuth!
+    removeCredentialsForIntegrationSystem(id: ID!): SystemAuth!
+}
+
+type Query {
+    integrationSystems(first: Int = 100, after: PageCursor): IntegrationSystemPage!
+    integrationSystem(id: ID!): IntegrationSystem
+}
+```
+Removing Integration System can be performed only when:
+- there is no Application Class from this Integration System
+- there is no Application managed by the Integration System 
+
+   
+2. Managing Application Class
+
+```
+input ApplicationClassInput {
+    name: String!
+    description: String
+    documentation: Document
+    integrationSystemID: String!
+    apiSpecs: [ApiSpecInput]
+    eventApiSpecs: [EventApiSpecInput]
+}
+
+input ApplicationInput {
+    ...
+    applicationClass: String
```

Comment:
IMO it will be be better to refer Application Class by human-readable name, not id. This impose of course that ApplicationClass.Name is unique. So maybe I will change this field to applicationClassName?